compiler/src/parsing/parsetree.re

```diff
@@ -658,8 +658,15 @@ type comment =
 
 /** The type for parsed programs */
 
+[@deriving (sexp, yojson)]
+type compile_flags = {
+  no_pervasives: bool,
+  runtime_mode: bool,
+};
+
 [@deriving (sexp, yojson)]
 type parsed_program = {
+  compile_flags,
```

Comment:
I'm not a huge fan of this being called `compile_flags` since there are other attributes we may add in the future that may not be related to flags. Ideally these are just `attributes` like anything else.